src/strings.c

```diff
@@ -880,20 +880,20 @@ string_count(char_u *haystack, char_u *needle, int ic)
     long	n = 0;
     char_u	*p = haystack;
     char_u	*next;
+    size_t	needlelen;
 
     if (p == NULL || needle == NULL || *needle == NUL)
 	return 0;
 
+    needlelen = STRLEN(needle);
```

Comment:
```suggestion
    size_t needlelen = STRLEN(needle);
```

and remove the definition in line 883 then.

Reply:
Yep.